dart/lib/src/io_client.dart

```diff
@@ -16,6 +17,7 @@ class SentryIOClient extends SentryClient {
   /// Instantiates a client using [SentryOptions]
   factory SentryIOClient(SentryOptions options) {
     options.sdk ??= Sdk(name: sdkName, version: sdkVersion);
+    options.httpClient ??= Client();
```

Comment:
this could be done on the `SentryClient` base class right?
the `Client.createClient` factory already creates a `IOClient` or `BrowserClient` based on dart:io availability

Reply:
each client needs its own httpClient. so if we move it to sentryClient, we'll need to add a conditional import `pkg:http/browser_client.dart' if(io)`pkg:http/http.dart' and a condition `isWeb ? BrowserClient() : Client()`
Do you think it's better ?

Reply:
as I understand, using an interface is platform agnostic, look at https://pub.dev/documentation/http/latest/http/Client/Client.html
unless I didn't get it right, the `conditional import` is already done on this Factory

Reply:
since 0.12.0 https://pub.dev/packages/http/changelog via https://github.com/dart-lang/http/pull/198

Reply:
thanks 👍

Reply:
✅